pkg/controller/external_nofork.go

```diff
@@ -543,15 +543,14 @@ func (n *noForkExternal) Observe(ctx context.Context, mg xpresource.Managed) (ma
 
 // sets the external-name on the MR. Returns `true`
 // if the external-name of the MR has changed.
-func (n *noForkExternal) setExternalName(mg xpresource.Managed, newState *tf.InstanceState) (bool, error) {
-	if newState.ID == "" {
+func (n *noForkExternal) setExternalName(mg xpresource.Managed, stateValueMap map[string]interface{}) (bool, error) {
+	id, ok := stateValueMap["id"]
+	if !ok {
```

Comment:
To be precise, we also need to make sure that the Terraform ID is not an empty string, i.e., it may be in the map but its value could be the empty string:
```suggestion
	if !ok || id.(string) == "" {
```
, with the assumption that if the `id` key exists, its value is a string.

Reply:
Done!